packages/shopping/src/authentication-strategies/jwt-strategy.ts

```diff
@@ -0,0 +1,50 @@
+// Copyright IBM Corp. 2019. All Rights Reserved.
+// Node module: @loopback/authentication
+// This file is licensed under the MIT License.
+// License text available at https://opensource.org/licenses/MIT
+
+import {inject} from '@loopback/context';
+import {HttpErrors, Request} from '@loopback/rest';
+import {AuthenticationStrategy, UserProfile} from '@loopback/authentication';
+import {TokenServiceBindings} from '../keys';
+import {JWTService} from '../services/jwt-service';
+
+export class JWTAuthenticationStrategy implements AuthenticationStrategy {
+  name: string = 'jwt';
+
+  constructor(
+    @inject(TokenServiceBindings.TOKEN_SERVICE)
+    public tokenService: JWTService,
```

Comment:
I think the type here should be [`TokenService`](https://github.com/strongloop/loopback-next/blob/master/packages/authentication/src/services/token.service.ts#L11)

Reply:
@jannyHou , It can be either, but I will change it to the interface instead of the implementation.

Reply:
@emonddr As we discussed earlier, feel free to ignore the comment about changing the interface :)

Reply:
@jannyHou , I decided to go with the interfaces as you suggested.